lib/butter.js

```diff
@@ -1,81 +1,114 @@
-        // detect if library is called in server side
-        if (typeof window === 'undefined'){
-          headers['Accept-Encoding'] = 'gzip'
-        }
-        var conn = axios.create({
-          baseURL: 'https://api.buttercms.com/v2',
-          timeout: timeout,
-          headers: headers
-        });
-
-        if (axiosHook) {
-          axiosHook(conn)
-        }
-
-        var params = params || {};
 
-        // Add api token to query string
-        params.auth_token = apiToken;
+    if (!(this instanceof Butter)) {
+        return new Butter(apiToken, options);
+    }
 
-        // Append &test=1&preview=1 query strings
-        if (testMode) {
-          params.test = 1
-          params.preview = 1
-        }
+    return Butter
+};
+
+/**
+ * The Butter function is the main entry point for interacting with the ButterCMS API.
+ * It takes an API token and an options object as parameters, and returns an object
+ * that provides methods for interacting with different types of resources in the API.
+ * 
+ * @param {string} apiToken - The API token for the ButterCMS service.
+ * @param {Object} options - Configuration options for the ButterCMS service.
+ * @property {string} options.cache - The cache mode for requests (default is "default").
+ * @property {Function} options.onError - A function to be called when an error occurs during a request.
+ * @property {Function} options.onRequest - A function to be called just before a request is made.
+ * @property {Function} options.onResponse - A function to be called when a response is received.
+ * @property {boolean} options.testMode - Whether to use test mode or live mode (default is false, which means live mode).
+ * @property {number} options.timeout - The timeout for requests in milliseconds (default is 3000ms).
+ * @returns {Object} An object that provides methods for interacting with the ButterCMS API.
+ */
+function Butter (apiToken, options ) {
+    const {
+        cache = "default",
+        onError = null,
+        onRequest = null, 
+        onResponse = null, 
+        testMode = false, 
+        timeout = 3000, 
+    } = options;
+
+    const config = {
+        apiToken,
+        cache,
+        onError,
+        onRequest,
+        onResponse,
+        testMode,
+        timeout,
+    }
 
-        return conn.get(url, {params: params})
-      }
+    return {
+        version: BUTTER_CLIENT_VERSION,
+        ...mapResources(resources, config)
     }
-  }
 }
 
-module.exports = Butter;
+/**
+ * Maps resource functions to their respective keys in lowercase.
+ * 
+ * @param {Object} resources - An object containing resource functions.
+ * @param {Object} config - The configuration object to pass to each resource function.
+ * @returns {Object} An object with the same keys as `resources`, but in lowercase, and values as the result of calling the respective function with `config`.
+ */
+function mapResources (resources, config) {
```

Comment:
Why couldn't we make it lowercase in the first place?

Reply:
was just keeping with the convention of the earlier versions. I can change it to lowercase if you think it doesnt matter?

I think the original devs thought of each resource as a class or component (like how we name in Vue), so they'd be Pascal-cased, but people aren't used to `butter.Page` so they lowercased them. 

What do you think?

Reply:
Let's leave it as is